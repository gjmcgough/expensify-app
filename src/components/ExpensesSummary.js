import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

const ExpensesSummary = (props) => {
  console.log("COUNT: " + props.expensesCount);
  const expenseWord = props.expensesCount > 1 ? 'expenses' : 'expense';
  const formattedExpensesTotal = numeral(props.expensesTotal / 100).format('$0,0.00');
  const totalExpenses = props.expensesCount;
  return (
    <div>
      <h2>Viewing {totalExpenses} {expenseWord} totalling {formattedExpensesTotal}.</h2>
    </div>
  );
};

export default ExpensesSummary;
