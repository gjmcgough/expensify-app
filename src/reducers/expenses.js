//EXPENSES Reducer
const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  // console.log('expenseReducer: received action', action);
  switch (action.type){
    case 'ADD_EXPENSE':
      return [
        ...state,
        action.expense
      ];
    case 'REMOVE_EXPENSE':
      return state.filter((expense) => expense.id !== action.id);
    case 'EDIT_EXPENSE':
      return state.map((expense) => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updates
          }
        } else {
          return expense;
        };
      });
    case 'SET_EXPENSES':
      return action.expenses;
    default:
    // console.log('expenseReducer: unknown action type', action.type);
      return state;
  }
};
