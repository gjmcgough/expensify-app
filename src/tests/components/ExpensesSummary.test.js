import React from 'react';
import { shallow } from 'enzyme';
import ExpensesSummary from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

let expensesCount, expensesTotal, wrapper;

beforeEach(() => {
  expensesCount = jest.fn();
  expensesTotal = jest.fn();
  wrapper = shallow(
    <ExpensesSummary
      expenses={expenses}
      expensesCount={expensesCount}
      expensesTotal={expensesTotal}
    />);
});

test('should render single expense singularily' , () => {

  expect(wrapper).toMatchSnapshot();
});

test('should render multiple expenses with plural expenses' , () => {
  expect(wrapper).toMatchSnapshot();
});
