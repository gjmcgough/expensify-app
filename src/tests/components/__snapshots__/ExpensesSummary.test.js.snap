// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render multiple expenses with plural expenses 1`] = `
<div>
  <h3>
    Viewing 
     expense totalling 
  </h3>
</div>
`;

exports[`should render single expense singularily 1`] = `
<div>
  <h3>
    Viewing 
     expense totalling 
  </h3>
</div>
`;
